lib/oxsdatatypes/src/date_time.rs

```diff
@@ -1766,7 +1769,35 @@ fn since_unix_epoch() -> Result<Duration, DateTimeError> {
     ))
 }
 
-#[cfg(not(all(target_family = "wasm", target_os = "unknown")))]
+#[cfg(all(feature = "custom-now", any(target_os = "unknown", target_os = "none")))]
+static GLOBAL_NEW: atomic::AtomicPtr<fn() -> Result<Duration, DateTimeError>> =
+    atomic::AtomicPtr::new(ptr::null_mut());
+
+/// Defines an implementation for fetching the current time.
+///
+/// Only enabled if there is no reachable operating system (target_os is unknown or none).
+#[cfg(all(feature = "custom-now", any(target_os = "unknown", target_os = "none")))]
+pub fn register_custom_now(now: fn() -> Result<Duration, DateTimeError>) {
+    GLOBAL_NEW.store(Box::leak(Box::new(now)), atomic::Ordering::Release);
```

Comment:
Do you really need acquire-release ordering guarantees between calls to `register_custom_now` and `since_unix_epoch`? Wouldn't `Ordering::Relaxed` suffice?

Reply:
(It also looks strange to box up the function pointer. But I suspect you are considering platforms where function and data pointers cannot be interchanged? Otherwise `AtomicPtr<()>` and transmuting the function pointer to `*mut ()` would probably suffice?)

Reply:
> Wouldn't Ordering::Relaxed suffice?

I believe there might be cases where the caller of `register_custom_now` sets some global variables used by the `now` function just before calling `register_custom_now` so it might be good to make sure they are properly accessible from the thread calling `now`. But I might be wrong, I am not an expert on memory ordering.

> But I suspect you are considering platforms where function and data pointers cannot be interchanged?

Not consciously. I just wanted a correct and easy to implement approach. In Oxigraph practical usages outside profiling, the `new` method is called only at most once per SPARQL query so I don't care much about performances. But being able to work on such platforms is nice. My main goal is to allow the author of [this fork](https://github.com/omnia-network/ic-oxigraph) to use upstream Oxigraph.